docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

```diff
@@ -0,0 +1,508 @@
+
+```go
+frame.Meta = &data.FrameMeta{
+	PreferredVisualization: visType,
+}
+```
+
+Example of constructing a data frame with specific meta information in `typescript`:
+
+> Note: Should be used if you are creating data frames on frontend in typescript code.
+
+```ts
+const result = new MutableDataFrame({
+    fields: [...],
+    meta: {
+        preferredVisualisationType: 'logs',
+    },
+});
+```
+
+### Color coded log levels
+
+> To be implemented in logs data frame as field
+
+Color coded [log level]({{< relref "../../../../explore/logs-integration/#log-level" >}}) is displayed at the beginning of the log line. It improves log visualization, allowing users to quickly assess the severity of log entries and facilitating effective log analysis and troubleshooting. Log level is determined from `severity` data frame field. In a case when it does not include `severity` field, Grafana tries to evaluate level based on the content of the log line. In a case it is not possible to do so, the log level is set to `unknown`.
+
+Refer to [Logs data frame format](#logs-data-frame-format) for more information.
+
+### Copy link to log line
+
+> To be implemented in logs data frame as field
+
+[Copy link to log line]({{< relref "../../../../explore/logs-integration/#copy-link-to-log-line" >}}) is a feature that allows you to generate a link to a specific log line for easy sharing or reference. This feature is supported for data sources that produce log data frames with `id` field.
+
+In a case where original database does not return `id` field, you can implement one within data source. For instance, in Loki data source, a combination of nanosecond timestamp, labels, and the content of the log line is used to create a unique id. On the other hand, Elasticsearch returns an `_id` field that is unique for the specified index. In such cases, to ensure uniqueness, both the `index name` and `_id` are used to create a unique id.
+
+Refer to [Logs data frame format](#logs-data-frame-format) for more information.
+
+### Log result meta information
+
+> To be implemented in logs data frame as field and/or meta
+
+[Log result meta information]({{< relref "../../../../explore/logs-integration/#log-result-meta-information" >}}) can be used to communicate information about logs result to the user. Information that can be shared with user:
+
+- **Count of received logs vs limit** - It displays the count of received logs compared to the specified limit. Data frame needs to have "limit" meta information with number of requested log lines.
+- **Error**: Displays possible error in your log results. Data frame needs to have "error" meta information.
+- **Common labels**: Labels, or attributes that are the same for all displayed log lines are shown as meta information. This feature is supported for data sources that produce log data frames with `attributes` field. Refer to [Logs data frame format](#logs-data-frame-format) for more information.
```

Comment:
i would recommend adding some explanation that for this (common labels) functionality nothing special has to be done by the datasource plugin..
maybe some phrasing like "calculated automatically by the logs panel" or something like that.